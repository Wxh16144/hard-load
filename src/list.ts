import { winPath } from './utils';

const CWD = process.cwd();
const NODE_MODULES = winPath(CWD, 'node_modules');

const ENABLED_OUTPUT = process.env.CI ? false : String(
  process.env.ENABLED_OUTPUT
  ?? process.env.WITH_OUTPUT
  ?? process.env.ENABLE_OUTPUT
  ?? process.env.OUTPUT
  ?? process.env.O
  ?? false
).toLowerCase() !== 'false';

export default [
  // cache
  winPath(NODE_MODULES, '.cache'),

  // vite
  winPath(NODE_MODULES, '.vite'),

  // umi
  winPath(
    CWD,
    `.${process.env.FRAMEWORK_NAME || 'umi'}`,
    "**"
  ),

  // dumi
  winPath(CWD, '.dumi/tmp-*',),

  // test coverage
  winPath(CWD, '.coverage', '**'),

  // compile output
  ...(
    ENABLED_OUTPUT
      ? [
        winPath(CWD, '{es,lib,dist}', '**'),
      ]
      : []
  )
]
