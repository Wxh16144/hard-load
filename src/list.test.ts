import path from 'path'
import { expect, describe, it } from 'vitest'
import list from './list'

describe('list', () => {
  it('matches the snapshot', () => {
    const nextList = list.map((item) => {
      return path.relative(process.cwd(), item)
    })

    expect(nextList).toMatchSnapshot()
  })
})