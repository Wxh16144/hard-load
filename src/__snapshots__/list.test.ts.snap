// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`list > matches the snapshot 1`] = `
[
  "node_modules/.cache",
  "node_modules/.vite",
  ".umi/**",
  ".dumi/tmp-*",
  ".coverage/**",
]
`;
